### ISSUE TITLE
Handle gesture collision between full-screen drawer drag and cards

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#8235]
Very interesting! Thanks for testing and reporting @TowelSniffer - I am not surprised there is an interaction issue and I appreciate the link to a use case that shows the issue.

We can 
- revert (always a possibility)
- see if we can make it an advanced feature toggle (I would say default to off since interaction problems like this are difficult to describe)
- see if there is a way to make it coexist (the best option if possible)

I'll open a separate issue for this and tag it for 2.15 so we don't cause regressions for people

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/issues/8071#issuecomment-799676820_

**Comment by haran2248** [comment]
Can I work on this?
Also can you upload a screenshot of the issue?

**Comment by TowelSniffer** [comment]
@haran2248, To describe the issue, with [this change](https://github.com/ankidroid/Anki-Android/pull/8071#issuecomment-799676820), when a user swipes right, the drawer opens as you see in the video. The issue I had is that some of my templates require [Dragging](https://ankiweb.net/shared/info/1082754005)/[Drawing](https://ankiweb.net/shared/info/379802892), and you can no longer swipe right on them as the drawer will open. 

Basically having the drawer open anytime a user swipes right on their cards can be problematic. 

https://user-images.githubusercontent.com/52064198/111232136-69427c00-85e2-11eb-9f44-6eb9e2bd5c32.mp4

**Comment by HuixingWong** [comment]
I think I know what's going on because the card is a WebView, and it is easy to fix this

**Comment by haran2248** [comment]
> @haran2248, To describe the issue, with [this change](https://github.com/ankidroid/Anki-Android/pull/8071#issuecomment-799676820), when a user swipes right, the drawer opens as you see in the video. The issue I had is that some of my templates require [Dragging](https://ankiweb.net/shared/info/1082754005)/[Drawing](https://ankiweb.net/shared/info/379802892), and you can no longer swipe right on them as the drawer will open.
> 
> Basically having the drawer open anytime a user swipes right on their cards can be problematic.
> 
>  110728693-71a05d00-8258-11eb-807f-678744002736.mp4

Ohh i got it now!So basically in such fragments/activities we can try to disable the navigationDrawer with swiping left?Is that a correct approach?Do you have any other ways to solve this problem?

**Comment by HuixingWong** [comment]
> > @haran2248, To describe the issue, with [this change](https://github.com/ankidroid/Anki-Android/pull/8071#issuecomment-799676820), when a user swipes right, the drawer opens as you see in the video. The issue I had is that some of my templates require [Dragging](https://ankiweb.net/shared/info/1082754005)/[Drawing](https://ankiweb.net/shared/info/379802892), and you can no longer swipe right on them as the drawer will open.
> > Basically having the drawer open anytime a user swipes right on their cards can be problematic.
> > 110728693-71a05d00-8258-11eb-807f-678744002736.mp4
> 
> Ohh i got it now!So basically in such fragments/activities we can try to disable the navigationDrawer with swiping left?Is that a correct approach?Do you have any other ways to solve this problem?

I have a better solution I'll push later

**Comment by haran2248** [comment]
> > > @haran2248, To describe the issue, with [this change](https://github.com/ankidroid/Anki-Android/pull/8071#issuecomment-799676820), when a user swipes right, the drawer opens as you see in the video. The issue I had is that some of my templates require [Dragging](https://ankiweb.net/shared/info/1082754005)/[Drawing](https://ankiweb.net/shared/info/379802892), and you can no longer swipe right on them as the drawer will open.
> > > Basically having the drawer open anytime a user swipes right on their cards can be problematic.
> > > 110728693-71a05d00-8258-11eb-807f-678744002736.mp4
> > 
> > 
> > Ohh i got it now!So basically in such fragments/activities we can try to disable the navigationDrawer with swiping left?Is that a correct approach?Do you have any other ways to solve this problem?
> 
> I have a better solution I'll push later

Can you discuss your solution?

**Comment by HuixingWong** [comment]
> > > > @haran2248, To describe the issue, with [this change](https://github.com/ankidroid/Anki-Android/pull/8071#issuecomment-799676820), when a user swipes right, the drawer opens as you see in the video. The issue I had is that some of my templates require [Dragging](https://ankiweb.net/shared/info/1082754005)/[Drawing](https://ankiweb.net/shared/info/379802892), and you can no longer swipe right on them as the drawer will open.
> > > > Basically having the drawer open anytime a user swipes right on their cards can be problematic.
> > > > 110728693-71a05d00-8258-11eb-807f-678744002736.mp4
> > > 
> > > 
> > > Ohh i got it now!So basically in such fragments/activities we can try to disable the navigationDrawer with swiping left?Is that a correct approach?Do you have any other ways to solve this problem?
> > 
> > 
> > I have a better solution I'll push later
> 
> Can you discuss your solution?

WebView's scroll event is different with other view,  can solve this problem on webView's onOverScrolled method
